libjolie/src/main/java/jolie/lang/parse/ast/OutputPortInfo.java

```diff
@@ -52,15 +49,29 @@ public void accept( OLVisitor visitor ) {
 		visitor.visit( this );
 	}
 
-	public String protocolId() {
-		return protocolId;
+	public OLSyntaxNode protocol() {
+		return protocol;
 	}
 
-	public OLSyntaxNode protocolConfiguration() {
-		return protocolConfiguration;
+	/**
+	 * @return an abstract node with defines protocol id this port is considering
+	 */
+	public OLSyntaxNode protocolId() {
+		OLSyntaxNode p = this.protocol;
+		if( p instanceof SumExpressionNode ) {
+			// a non optimized protocol node
+			OLSyntaxNode prodNode = ((SumExpressionNode) protocol).operands().iterator().next().value();
+			p = ((ProductExpressionNode) prodNode).operands().iterator().next().value();
+		}
+
+		if( p instanceof InlineTreeExpressionNode ) {
+			return ((InlineTreeExpressionNode) p).rootExpression();
```

Comment:
This is different than the one in `InputPortInfo`. Why? Should they?
If not, you might wanna extract this code to a static method in `PortInfo` and call it from both this method and the one in InputPortInfo.